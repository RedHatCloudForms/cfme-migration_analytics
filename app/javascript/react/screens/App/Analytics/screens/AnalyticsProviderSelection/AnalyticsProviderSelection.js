import React from 'react';
import PropTypes from 'prop-types';
import { TypeAheadSelect, Button, Form, Icon } from 'patternfly-react';

const AnalyticsProviderSelection = ({
  summaryData,
  selectedProviders,
  selectProvidersAction,
  detailedDataSelected,
  selectDetailedDataAction,
  onContinueClick,
  onCancelClick
}) => (
  <Form>
    <p>{__('Select providers for which inventory data will be collected.')}</p>
    <TypeAheadSelect
      inputProps={{ id: 'providers-select' }}
      options={summaryData.providers}
      selected={selectedProviders}
      onChange={providers => selectProvidersAction(providers)}
      labelKey="name"
      placeholder={__('Select...')}
      emptyLabel={__('No matches found.')}
      multiple
      clearButton
      highlightOnlyResult
      selectHintOnEnter
    />
    <br />

    <Form.FormGroup className="detailed-data-selection">
      <Form.ControlLabel>{__('Inventory data needed')}</Form.ControlLabel>
      <Form.Radio
        checked={!detailedDataSelected}
        onChange={event => selectDetailedDataAction(event.target.value !== 'on')}
      >
        {__('Basic data')}
        <ul>
          <li>{__('High level information including number of VMs and hypervisors.')}</li>
          <li>{__('Necessary to calculate an initial migration savings estimate.')}</li>
          <li>{__('Data may or may not include SmartState Analysis results.')}</li>
        </ul>
      </Form.Radio>
      <Form.Radio
        checked={detailedDataSelected}
        onChange={event => selectDetailedDataAction(event.target.value === 'on')}
      >
        {__('Detailed data')}
        <ul>
          <li>
            {__('Detailed analysis of individual VMs including host, operating system, files, and running services.')}
          </li>
          <li>{__('Necessary for creating workload inventory and workload summary reports.')}</li>
          <li>
            {__('Requires that SmartState Analysis be run on each VM. This can take several hours depending on the number of VMs in the selected providers.') /* prettier-ignore */}
          </li>
        </ul>
        {detailedDataSelected && (
          <div className="icon-with-content smart-state-info">
            <Icon type="pf" name="warning-triangle-o" className="p-aligned" />
            <p>
              {__('See product documentation for instructions to configure and run SmartState Analysis.')}
              <br />
              <a href="/miq_task">{__('Progress of Smart State Analysis tasks can be monitored on the Tasks page.')}</a>
            </p>
          </div>
        )}
      </Form.Radio>
    </Form.FormGroup>

    <div className="footer-buttons">
      <Button bsStyle="primary" onClick={onContinueClick} disabled={selectedProviders.length === 0}>
        {__('Continue')}
      </Button>
      <Button onClick={onCancelClick}>{__('Cancel')}</Button>
    </div>
  </Form>
);

const providerShape = PropTypes.shape({
  id: PropTypes.number,
  name: PropTypes.string
});

AnalyticsProviderSelection.propTypes = {
  summaryData: PropTypes.shape({
    providers: PropTypes.arrayOf(providerShape)
  }).isRequired,
  selectedProviders: PropTypes.arrayOf(providerShape).isRequired,
  selectProvidersAction: PropTypes.func.isRequired,
  detailedDataSelected: PropTypes.bool.isRequired,
  selectDetailedDataAction: PropTypes.func.isRequired,
  onContinueClick: PropTypes.func.isRequired,
  onCancelClick: PropTypes.func.isRequired
};

export default AnalyticsProviderSelection;
