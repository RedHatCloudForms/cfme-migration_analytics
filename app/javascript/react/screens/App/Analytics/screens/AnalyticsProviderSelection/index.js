import { connect } from 'react-redux';
import AnalyticsProviderSelection from './AnalyticsProviderSelection';

const mapStateToProps = ({
  migrationAnalytics: {
    analytics: { summaryData }
  }
}) => ({
  summaryData
});

export default connect(
  mapStateToProps,
  {}
)(AnalyticsProviderSelection);
