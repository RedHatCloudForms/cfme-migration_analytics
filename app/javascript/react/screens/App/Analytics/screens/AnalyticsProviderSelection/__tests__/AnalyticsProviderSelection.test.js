import React from 'react';
import { shallow } from 'enzyme';
import { TypeAheadSelect, Form } from 'patternfly-react';
import AnalyticsProviderSelection from '../AnalyticsProviderSelection';

describe('analytics provider selection screen', () => {
  const getBaseProps = () => ({
    summaryData: {
      providers: [{ id: 1, name: 'Provider 1' }, { id: 2, name: 'Provider 2' }]
    },
    selectedProviders: [],
    selectProvidersAction: jest.fn(),
    detailedDataSelected: false,
    selectDetailedDataAction: jest.fn(),
    onContinueClick: jest.fn(),
    onCancelClick: jest.fn()
  });

  test('renders correctly with basic data selected', () => {
    const component = shallow(<AnalyticsProviderSelection {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
    expect(component.find('.smart-state-info')).toHaveLength(0);
  });

  test('renders radio buttons section correctly with detailed data selected', () => {
    const props = {
      ...getBaseProps(),
      detailedDataSelected: true
    };
    const component = shallow(<AnalyticsProviderSelection {...props} />);
    expect(component.find('.detailed-data-selection')).toMatchSnapshot();
    expect(component.find('.smart-state-info')).toHaveLength(1);
  });

  test('calls action correctly when changing provider selection', () => {
    const props = getBaseProps();
    const component = shallow(<AnalyticsProviderSelection {...props} />);
    component
      .find(TypeAheadSelect)
      .props()
      .onChange(['mock', 'providers']);
    expect(props.selectProvidersAction).toHaveBeenCalledWith(['mock', 'providers']);
  });

  test('calls action correctly when selecting basic data', () => {
    const props = getBaseProps();
    const component = shallow(<AnalyticsProviderSelection {...props} />);
    component
      .find(Form.Radio)
      .at(0)
      .props()
      .onChange({ target: { value: 'on' } });
    expect(props.selectDetailedDataAction).toHaveBeenCalledWith(false);
  });

  test('calls action correctly when deselecting basic data', () => {
    const props = getBaseProps();
    const component = shallow(<AnalyticsProviderSelection {...props} />);
    component
      .find(Form.Radio)
      .at(0)
      .props()
      .onChange({ target: { value: 'off' } });
    expect(props.selectDetailedDataAction).toHaveBeenCalledWith(true);
  });

  test('calls action correctly when selecting detailed data', () => {
    const props = getBaseProps();
    const component = shallow(<AnalyticsProviderSelection {...props} />);
    component
      .find(Form.Radio)
      .at(1)
      .props()
      .onChange({ target: { value: 'on' } });
    expect(props.selectDetailedDataAction).toHaveBeenCalledWith(true);
  });

  test('calls action correctly when deselecting detailed data', () => {
    const props = getBaseProps();
    const component = shallow(<AnalyticsProviderSelection {...props} />);
    component
      .find(Form.Radio)
      .at(1)
      .props()
      .onChange({ target: { value: 'off' } });
    expect(props.selectDetailedDataAction).toHaveBeenCalledWith(false);
  });
});
