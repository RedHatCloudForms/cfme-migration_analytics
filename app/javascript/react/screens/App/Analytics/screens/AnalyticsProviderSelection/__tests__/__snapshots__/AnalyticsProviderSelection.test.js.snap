// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics provider selection screen renders correctly with basic data selected 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <p>
    Select providers for which inventory data will be collected.
  </p>
  <TypeaheadContainer(WrappedTypeahead)
    a11yNumResults={[Function]}
    a11yNumSelected={[Function]}
    align="justify"
    allowNew={false}
    autoFocus={false}
    caseSensitive={false}
    clearButton={true}
    defaultInputValue=""
    defaultOpen={false}
    defaultSelected={Array []}
    disabled={false}
    dropup={false}
    emptyLabel="No matches found."
    filterBy={Array []}
    flip={false}
    highlightOnlyResult={true}
    id="providers-select-typeahead"
    ignoreDiacritics={true}
    inputProps={
      Object {
        "id": "providers-select",
      }
    }
    isInvalid={false}
    isLoading={false}
    isValid={false}
    labelKey="name"
    maxResults={100}
    minLength={0}
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInputChange={[Function]}
    onKeyDown={[Function]}
    onPaginate={[Function]}
    options={
      Array [
        Object {
          "id": 1,
          "name": "Provider 1",
        },
        Object {
          "id": 2,
          "name": "Provider 2",
        },
      ]
    }
    paginate={true}
    paginationText="Display additional results..."
    placeholder="Select..."
    selectHintOnEnter={true}
    selected={Array []}
  />
  <br />
  <FormGroup
    bsClass="form-group"
    className="detailed-data-selection"
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      Inventory data needed
    </ControlLabel>
    <Radio
      bsClass="radio"
      checked={true}
      disabled={false}
      inline={false}
      onChange={[Function]}
      title=""
    >
      Basic data
      <ul>
        <li>
          High level information including number of VMs and hypervisors.
        </li>
        <li>
          Necessary to calculate an initial migration savings estimate.
        </li>
        <li>
          Data may or may not include SmartState Analysis results.
        </li>
      </ul>
    </Radio>
    <Radio
      bsClass="radio"
      checked={false}
      disabled={false}
      inline={false}
      onChange={[Function]}
      title=""
    >
      Detailed data
      <ul>
        <li>
          Detailed analysis of individual VMs including host, operating system, files, and running services.
        </li>
        <li>
          Necessary for creating workload inventory and workload summary reports.
        </li>
        <li>
          Requires that SmartState Analysis be run on each VM. This can take several hours depending on the number of VMs in the selected providers.
        </li>
      </ul>
    </Radio>
  </FormGroup>
  <div
    className="footer-buttons"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={true}
      onClick={[MockFunction]}
    >
      Continue
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      Cancel
    </Button>
  </div>
</Form>
`;

exports[`analytics provider selection screen renders radio buttons section correctly with detailed data selected 1`] = `
<FormGroup
  bsClass="form-group"
  className="detailed-data-selection"
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    Inventory data needed
  </ControlLabel>
  <Radio
    bsClass="radio"
    checked={false}
    disabled={false}
    inline={false}
    onChange={[Function]}
    title=""
  >
    Basic data
    <ul>
      <li>
        High level information including number of VMs and hypervisors.
      </li>
      <li>
        Necessary to calculate an initial migration savings estimate.
      </li>
      <li>
        Data may or may not include SmartState Analysis results.
      </li>
    </ul>
  </Radio>
  <Radio
    bsClass="radio"
    checked={true}
    disabled={false}
    inline={false}
    onChange={[Function]}
    title=""
  >
    Detailed data
    <ul>
      <li>
        Detailed analysis of individual VMs including host, operating system, files, and running services.
      </li>
      <li>
        Necessary for creating workload inventory and workload summary reports.
      </li>
      <li>
        Requires that SmartState Analysis be run on each VM. This can take several hours depending on the number of VMs in the selected providers.
      </li>
    </ul>
    <div
      className="icon-with-content smart-state-info"
    >
      <Icon
        className="p-aligned"
        name="warning-triangle-o"
        type="pf"
      />
      <p>
        See product documentation for instructions to configure and run SmartState Analysis.
        <br />
        <a
          href="/miq_task"
        >
          Progress of SmartState Analysis tasks can be monitored on the Tasks page.
        </a>
      </p>
    </div>
  </Radio>
</FormGroup>
`;
